docs/tutorial/file_handling.md

```diff
@@ -0,0 +1,471 @@
+
+  def run: IO[Unit] =
+    for
+      exists <- FilesOs.exists(annoyingFile)
+      _ <- FilesOs.delete(annoyingFile).whenA(exists)
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+Or even better, use the convenience method `deleteIfExists`:
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      deleted <- annoyingFile.deleteIfExists
+      _ <- IO.println(s"Are they reaching out? $deleted")
+    yield ()
+
+end Deleting
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val annoyingFile = Path("desktop/extend_your_car_warranty_now.ad")
+
+  def run: IO[Unit] =
+    for
+      deleted <- FilesOs.deleteIfExists(annoyingFile)
+      _ <- IO.println(s"Are they reaching out? $deleted")
+    yield ()
+
+end Deleting
+```
+
+@:@
+
+It will return a boolean indicating whether the file has been deleted (also deletes directories).
+
+Lastly, you may want to delete not one but multiple files and directories, here is when the `deleteDirectorires` comes handy, as it will delete all the files and directories recursively (similar to `rm -r`):
+
+**Before:**
+
+```
+/My files
+├── /non empty folder
+│   ├── 3751c91b_2024-06-16_7.csv
+│   ├── Screenshot 2024-06-16 210523.png
+│   ├── /downloaded
+│   │   ├── /on_internet
+│   │   │   └── Unconfirmed 379466.crdownload
+│   │   └── ubuntu-24.04-desktop-amd64.iso
+│   └── /spark
+│       ├── output0-part-r-00000.nodes
+│       └── output1-part-r-00000.nodes
+│ 
+├── /dont delete
+│ 
+```
+
+@:select(api-style)
+
+@:choice(syntax)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.syntax.path.*
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val nonEmptyFolder = Path("downloads/non empty folder")
+
+  def run: IO[Unit] = nonEmptyFolder.deleteRecursively
+
+end Deleting
+```
+
+@:choice(static)
+
+```scala mdoc:compile-only
+import io.chrisdavenport.shellfish.FilesOs
+import fs2.io.file.Path
+import cats.effect.{IO, IOApp}
+
+object Deleting extends IOApp.Simple:
+
+  val nonEmptyFolder = Path("downloads/non empty folder")
+
+  def run: IO[Unit] = FilesOs.deleteRecursively(nonEmptyFolder)
+
+end Deleting
+```
+
+@:@
+
+**After:**
+
+```
+/My files
+├── /dont delete
+│  
+```
+
+### Exercise
+
+You are tired of people abusing the FTP server to upload long pirated movies. So you decide to implement a method that checks if a file exceeds a certain size, and if so, automatically deletes it (hint: check out the `size` method!). Return `true` if the file was deleted, `false` otherwise:
```

Comment:
```suggestion
You are tired of people abusing the FTP server to upload enormous files, so you decide to implement a method that checks if a file exceeds a certain size, and if so, automatically delete it (hint: check the `size` method). The method must return `true` if the file has been deleted, `false` otherwise.
```